### ISSUE TITLE
Improve path inference

### ISSUE THREAD

**Issue by jenseng** [jenseng/eslint-plugin-remix-react-routes#5]
While the current rules work well enough for simple string and template literals, they can hit false positives with other expressions. E.g. consider something like this:

```tsx
const BASE_URL = "http://example.com";

return <a href={`${BASE_URL}/something`} target="_blank">Some site</a>
```

This gets incorrectly flagged by [use-link-for-routes](https://github.com/jenseng/eslint-plugin-remix-react-routes/tree/main/src/rules/use-link-for-routes) since it doesn't know what `BASE_URL` is. While we can't solve every case statically, we probably can solve some basic stuff, e.g. const string literals and the like. We might even be able to leverage `@typescript-eslint` to resolve const strings that come from elsewhere, since they should be known to the type system.

**Comment by jenseng** [comment]
For the specific example given, a simpler fix could be for that rule to check if there's a `target="_blank"` attribute, and then just allow anything (since it's going to open in a new tab/window, making `<Link>` inappropriate)

That said, it would still be good to improve path inference generally

**Comment by jenseng** [comment]
One other thought here, if the attribute is a template literal that starts with a substitution, we could just 🤷 and not infer a string value, just as we do with other non string/template literal attribute values.
